studio-docs/source/sharing-graphs.mdx

```diff
@@ -0,0 +1,498 @@
+const EXPLORER_SUBSCRIPTION_REQUEST = "ExplorerSubscriptionRequest";
+const EXPLORER_SUBSCRIPTION_RESPONSE = "ExplorerSubscriptionResponse";
+const SUBSCRIPTION_TERMINATION = "ExplorerSubscriptionTermination";
+```
+
+</MultiCodeBlock>
+
+#### 2. Set up communication
+
+Next, let's define a function that sets up communication between a webpage and the embedded Explorer's `iframe`. This function defines a callback that will execute whenever `window.postMessage` is called by the `iframe`.
+
+<MultiCodeBlock>
+
+```ts:title=setupEmbedRelay.ts
+export function setUpEmbedRelay() {
+
+  // Callback definition
+  const onPostMessageReceived = (
+    event: MessageEvent<{
+      name?: string;
+      operation?: string;
+      operationName?: string;
+      variables?: string;
+      headers?: Record<string, string>;
+    }>
+  ) => {
+
+    // Obtain the iframe element by any applicable logic for your page.
+    // This obtains an element with ID `embedded-explorer`.
+    const embeddedExplorerIFrame =
+      (document.getElementById("embedded-explorer") as HTMLIFrameElement) ??
+      undefined;
+
+    // Check to see if the posted message indicates that the user is
+    // executing a query or mutation in the Explorer
+    const isQueryOrMutation =
+      "name" in event.data &&
+      event.data.name?.startsWith(`${EXPLORER_QUERY_MUTATION_REQUEST}:`);
+
+    // If the user is executing a query or mutation...
+    if (
+      isQueryOrMutation &&
+      event.data.name &&
+      event.data.operation
+    ) {
+
+      // Extract the operation's unique ID from the message's name
+      // (the ID is after the colon)
+      const operationId = event.data.name.split(":")[1];
+
+      // Extract the operation's other details from the event.data object
+      const { operation, operationName, variables, headers } = event.data;
+
+      // Execute the operation, providing all required details
+      // (we'll define executeOperation next)
+      executeOperation({
+        operation: event.data.operation,
+        operationName,
+        variables,
+        headers,
+        embeddedExplorerIFrame,
+        operationId,
+      });
+    }
+  };
+
+  // Execute our callback whenever window.postMessage is called
+  window.addEventListener("message", onPostMessageReceived);
+}
+```
+
+</MultiCodeBlock>
+
+Now we can import and call `setupEmbedRelay` on any page where we'll embed the Explorer. However, we still need to define the `executeOperation` function that's called by our callback.
+
+#### 3. Define operation execution logic
+
+Our webpages can use the `fetch` API to send operations to our GraphQL server. To do this, we need the URL of our server's GraphQL endpoint. Substitute it in the `executeOperation` function below, where indicated.
+
+<MultiCodeBlock>
+
+```ts:title=setupEmbedRelay.ts
+// Helper function that adds content-type: application/json
+// to each request's headers if not present
+function getHeadersWithContentType(
```

Comment:
Oh did you always have this here? sorry if I missed it last time!

Reply:
Ah no not at all! I added it thanks to your comment!